lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
@indirect Is the refactor you are suggesting necessary for the bugfix this PR hopes to address? Since line 39 in this file is different from the actual behavior in Rubygems that @drbrain mentioned, I was hesitant to encapsulate this as a `Bundler.rubygems` method.